perception/autoware_tensorrt_bevdet/src/alignbev_plugin.cu

```diff
@@ -0,0 +1,352 @@
+    }
+    count *= output_desc.shape[i];
+  }
+
+  switch (int(outputDesc[0].type)) {
+    case int(DataType::kFLOAT):
+      if (inputDesc[0].type == DataType::kFLOAT) {
+        // align : fp32, fp32;
+        align_bev_kernel<float, float><<<GET_BLOCKS(count), NUM_THREADS, 0, stream>>>(
+          count, reinterpret_cast<const float *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<float *>(outputs[0]),
+          output_desc);
+      } else {
+        // align : fp16, fp32
+        align_bev_kernel<__half, float><<<GET_BLOCKS(count), NUM_THREADS, 0, stream>>>(
+          count, reinterpret_cast<const __half *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<float *>(outputs[0]),
+          output_desc);
+      }
+      break;
+    case int(DataType::kHALF):
+      if (inputDesc[0].type == DataType::kFLOAT) {
+        // align : fp32, fp16
+        align_bev_kernel<float, __half><<<GET_BLOCKS(count), NUM_THREADS, 0, stream>>>(
+          count, reinterpret_cast<const float *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<__half *>(outputs[0]),
+          output_desc);
+      } else {
+        // align : fp16, fp16
+        align_bev_kernel<__half, __half><<<GET_BLOCKS(count), NUM_THREADS, 0, stream>>>(
+          count, reinterpret_cast<const __half *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<__half *>(outputs[0]),
+          output_desc);
+      }
+      break;
+    default:  // should NOT be here
+      RCLCPP_ERROR(rclcpp::get_logger("AlignBEVPlugin"), "\tUnsupported datatype!");
+  }
+
+  return 0;
+}
+
+void AlignBEVPlugin::destroy() noexcept
+{
+  delete this;
+  return;
+}
+
+void AlignBEVPlugin::configurePlugin(
+  const DynamicPluginTensorDesc * in, int32_t nbInputs, const DynamicPluginTensorDesc * out,
+  int32_t nbOutputs) noexcept
+{
+  return;
+}
+
+int32_t AlignBEVPlugin::initialize() noexcept
+{
+  return 0;
+}
+
+void AlignBEVPlugin::terminate() noexcept
+{
+  return;
+}
+
+size_t AlignBEVPlugin::getSerializationSize() const noexcept
+{
+  return sizeof(m_);
+}
+
+void AlignBEVPlugin::serialize(void * buffer) const noexcept
+{
+  memcpy(buffer, &m_, sizeof(m_));
+  return;
+}
+
+void AlignBEVPlugin::setPluginNamespace(const char * pluginNamespace) noexcept
+{
+  namespace_ = pluginNamespace;
+  return;
+}
+
+const char * AlignBEVPlugin::getPluginNamespace() const noexcept
+{
+  return namespace_.c_str();
+}
+
+const char * AlignBEVPlugin::getPluginType() const noexcept
+{
+  return ALIGN_PLUGIN_NAME;
+}
+
+const char * AlignBEVPlugin::getPluginVersion() const noexcept
+{
+  return ALIGN_PLUGIN_VERSION;
+}
+
+void AlignBEVPlugin::attachToContext(
+  cudnnContext * contextCudnn, cublasContext * contextCublas, IGpuAllocator * gpuAllocator) noexcept
+{
+  return;
+}
+
+void AlignBEVPlugin::detachFromContext() noexcept
+{
+  return;
+}
+
+// class AlignBEVPluginCreator
```

Comment:
Similar to the `AlignBEVPlugin` class, consider wrapping the `AlignBEVPluginCreator` class within the same project namespace to maintain consistency and avoid potential naming conflicts.